pkg/driver/node/node.go

```diff
@@ -56,28 +59,18 @@ var (
 
 // S3NodeServer is the implementation of the csi.NodeServer interface
 type S3NodeServer struct {
-	NodeID             string
-	Mounter            mounter.Mounter
-	credentialProvider *mounter.CredentialProvider
+	nodeID             string
+	mounter            mounter.Mounter
+	credentialProvider *credentialprovider.Provider
+	regionProvider     *regionprovider.Provider
+	kubernetesVersion  string
 }
 
-func NewS3NodeServer(nodeID string, mounter mounter.Mounter, credentialProvider *mounter.CredentialProvider) *S3NodeServer {
-	return &S3NodeServer{NodeID: nodeID, Mounter: mounter, credentialProvider: credentialProvider}
+func NewS3NodeServer(nodeID string, mounter mounter.Mounter, credentialProvider *credentialprovider.Provider, regionProvider *regionprovider.Provider, kubernetesVersion string) *S3NodeServer {
+	return &S3NodeServer{nodeID: nodeID, mounter: mounter, credentialProvider: credentialProvider, regionProvider: regionProvider, kubernetesVersion: kubernetesVersion}
 }
 
 func (ns *S3NodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRequest) (*csi.NodeStageVolumeResponse, error) {
-	volumeContext := req.GetVolumeContext()
-	if volumeContext[mounter.VolumeCtxAuthenticationSource] == mounter.AuthenticationSourcePod {
-		podID := volumeContext[mounter.VolumeCtxPodUID]
-		volumeID := req.GetVolumeId()
-		if podID != "" && volumeID != "" {
-			err := ns.credentialProvider.CleanupToken(volumeID, podID)
-			if err != nil {
-				klog.V(4).Infof("NodeStageVolume: Failed to cleanup token for pod/volume %s/%s: %v", podID, volumeID, err)
-			}
-		}
-	}
```

Comment:
This `NodeStageVolume` wasn't enabled at all – we need to add `STAGE_UNSTAGE_VOLUME` node capability to enable – and also this cleaning of tokens has been moved to `Unmount` method of mounter.